alua.c

```diff
@@ -483,10 +519,28 @@ int tcmu_transition_tgt_port_grp(struct tgt_port_grp *group, uint8_t new_state,
 	return SAM_STAT_GOOD;
 }
 
+static int tcmu_report_state(struct tcmu_device *dev,
+			     struct tgt_port_grp *group)
+{
+	struct tcmur_handler *rhandler = tcmu_get_runner_handler(dev);
+	int ret;
+
+	/* TODO: For ESX return remote ports */
+
+	if (!rhandler->has_lock)
+		return group->state;
+
+	ret = rhandler->has_lock(dev);
+	if (ret <= 0) {
+		return ALUA_ACCESS_STATE_STANDBY;
+	} else {
+		return ALUA_ACCESS_STATE_OPTIMIZED;
+	}
+}
+
 int tcmu_emulate_report_tgt_port_grps(struct tcmu_device *dev,
 				      struct list_head *group_list,
-				      struct tcmulib_cmd *cmd,
-				      tcmu_report_state_fn_t *report_fn)
+				      struct tcmulib_cmd *cmd)
```

Comment:
> No, I just personally think that runner is needed more than libtcmu. It would make dev/maintain life easier withtout attention to libtcmu.

Agree. Lol. We completely agree on something! :)

I will try to find some time to dig into all the past commits and contact the original users and open a PR to depreciate support.